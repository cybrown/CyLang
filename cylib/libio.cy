void extern putchar(int c);
void extern putbyte(byte c);
void extern putshort(short c);
void extern putint(int c);
void extern putlong(long c);
void extern putfloat(float f);
void extern putdouble(double d);
void extern puts(byte[] str);
void extern printf(byte[] str);

void print(byte c) {
	putbyte(c);
}

void print(short c) {
	putshort(c);
}

void print(int c) {
	putint(c);
}

void print(long c) {
	putlong(c);
}

void print(float f) {
	putfloat(f);
}

void print(double d) {
	putdouble(d);
}

void print(byte[] str) {
	printf(str);
}

void print(bool c) {
	print(c ? "true" : "false");
}
