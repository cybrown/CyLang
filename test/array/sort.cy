import libio;

void sort(int[] arr, int size) {
	int tmp;
	int i;
	bool flag = true;
	while (flag)
	{
		flag = false;
		for (i = 0; i < size-1; i += 1)
		{
			if (arr[i] > arr[i+1])
			{
				tmp = arr[i];
				arr[i] = arr[i+1];
				arr[i+1] = tmp;
				flag = true;
			}
		}
		i = 0;
	}
	return;
}

int[10] array;
array[0] = 51;
array[1] = 32;
array[2] = 83;
array[3] = 10;
array[4] = 42;
array[5] = 96;
array[6] = 35;
array[7] = 75;
array[8] = 25;
array[9] = 97;

int i;

for (i = 0; i < 10; i += 1)
{
	putint(array[i]);
	putchar(10);
}

putchar(10);
sort(array, 10);

for (i = 0; i < 10; i += 1)
{
	putint(array[i]);
	putchar(10);
}
