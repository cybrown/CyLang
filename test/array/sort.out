51
32
83
10
42
96
35
75
25
97

10
25
32
35
42
51
75
83
96
97
